lib/codecs/src/decoding/format/syslog.rs

```diff
@@ -143,6 +236,110 @@ fn resolve_year((month, _date, _hour, _min, _sec): IncompleteDate) -> i32 {
     }
 }
 
+fn insert_metatdata_fields_from_syslog(
```

Comment:
This makes me wonder if we want a helper similar to `insert_source|vector_metadata()` that would toggle between inserting on the event and inserting into metadata as needed.